ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -1299,13 +1285,38 @@ run tracers tracersExtra args argsExtra apps appsExtra = do
 #else
                  diInstallSigUSR1Handler = \_ -> pure ()
 #endif
-                 diNtnDomainResolver :: [DomainAccessPoint]
+
+             cmIPv4Address <- traverse (either (Snocket.getLocalAddr diNtnSnocket) pure)
+                                  (daIPv4Address args)
+             case cmIPv4Address of
+                  Just addr | Just IPv4Address <- socketAddressType addr
+                            -> pure ()
+                            | otherwise
+                            -> throwIO (UnexpectedIPv4Address addr)
+                  Nothing   -> pure ()
+
+             cmIPv6Address <- traverse (either (Snocket.getLocalAddr diNtnSnocket) pure)
+                                  (daIPv6Address args)
+             case cmIPv6Address of
+                    Just addr | Just IPv6Address <- socketAddressType addr
+                              -> pure ()
+                              | otherwise
+                              -> throwIO (UnexpectedIPv6Address addr)
+                    Nothing   -> pure ()
+
+             lookupReqs <- case (cmIPv4Address, cmIPv6Address) of
+                                   (Just _, Nothing) -> return LookupReqAOnly
+                                   (Nothing, Just _) -> return LookupReqAAAAOnly
+                                   (Just _, Just _)  -> return LookupReqAAndAAAA
+                                   _                 ->
+                                       throwIO (NoSocket :: Failure RemoteAddress)
+             let diNtnDomainResolver :: [DomainAccessPoint]
```

Comment:
I see that's probably because of `lookupReqs` that is need to constrcutr `ioDNSActions`.    We could redefine `diDNSActions` as `diDNSActions :: LookupReqs -> DNSActions`.

Reply:
Yes, I moved it there in order to be able to use it to construct `diDNSActions`.

Reply:
I still think that passing `diDNSActions :: LookupReqs -> DNSActions` and initialisation `DNSActions` in `runM` is cleaner.

Reply:
Will make that change.